server/verify_test.go

```diff
@@ -1065,3 +1191,62 @@ func TestVerifyAttestationWithSevSnp(t *testing.T) {
 		})
 	}
 }
+
+func TestVerifyAttestationWithTdx(t *testing.T) {
+
+	tdxEventLog := createTpm2EventLog(3)
+	rwc := test.GetSimulatorWithLog(t, tdxEventLog)
+	defer client.CheckedClose(t, rwc)
+
+	ak, err := client.AttestationKeyRSA(rwc)
+	if err != nil {
+		t.Fatalf("failed to generate AK: %v", err)
+	}
+	defer ak.Close()
+
+	nonce := []byte("super secret nonce")
+	altNonce := []byte("alternate secret nonce")
+	var nonce64 [64]byte
+	copy(nonce64[:], altNonce)
+	tdxTestDevice := tgtestclient.GetTdxGuest([]tgtest.TestCase{
+		{
+			Input: nonce64,
+			Quote: tgtestdata.RawQuote,
+		},
+	}, t)
+	defer tdxTestDevice.Close()
+	attestation, err := ak.Attest(client.AttestOpts{
+		Nonce:     nonce,
+		TEEDevice: &client.TdxDevice{Device: tdxTestDevice},
+		TEENonce:  nonce64[:],
+	})
+	if err != nil {
+		t.Fatalf("failed to attest: %v", err)
+	}
+
+	type testCase struct {
+		name    string
+		opts    VerifyOpts
+		wantErr string
+	}
+	tcs := []testCase{
+		{
```

Comment:
there should be failure tests (e.g., bad key, bad cert, bad nonce, etc.)

Reply:
Added a failure testcase for wrong TDX Attestation. Since go-tdx-guest has pending validate changes, I will add a testcase related to badNonce in the later PR.